mreg/api/v1/views_network_policy.py

```diff
@@ -0,0 +1,249 @@
+    queryset = NetworkPolicyAttribute.objects.all()
+    serializer_class = NetworkPolicyAttributeSerializer
+    permission_classes = (IsSuperOrNetworkAdminMember,)
+    filterset_class = NetworkPolicyAttributeFilterSet
+    ordering_fields = ("id",)
+
+
+    def create(self, request, *args, **kwargs):
+        name = request.data.get("name")
+        if not name:
+            raise exceptions.ValidationError("'name' is required.")
+        
+        try:
+            NetworkPolicyAttribute.objects.get(name=name)
+            raise ValidationError409(detail=f"NetworkPolicyAttribute with the name '{name}' already exists.")
+        except NetworkPolicyAttribute.DoesNotExist:
+            pass
+
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        with transaction.atomic():
+            network_policy_attribute = serializer.save()
+
+        headers = self.get_success_headers(serializer.data)
+        headers["Location"] = request.build_absolute_uri(
+            reverse(URL.NetworkPolicy.ATTRIBUTE_DETAIL, kwargs={"pk": network_policy_attribute.id})
+        )
+
+        return response.Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+class NetworkPolicyAttributeDetail(JSONContentTypeMixin, generics.RetrieveUpdateDestroyAPIView):
+    queryset = NetworkPolicyAttribute.objects.all()
+    serializer_class = NetworkPolicyAttributeSerializer
+    permission_classes = (IsSuperOrNetworkAdminMember,)
+
+
+class NetworkCommunityList(JSONContentTypeMixin, generics.ListCreateAPIView):
+    serializer_class = CommunitySerializer
+    permission_classes = (IsSuperOrNetworkAdminMember,)
+    filterset_class = CommunityFilterSet
+
+    def get_queryset(self):
+        network = self.kwargs.get("network")
+        return Community.objects.filter(network__network=network).order_by("id")
+
+    def create(self, request, *args, **kwargs):
+        network = self.kwargs.get("network")
+
+        if not network:
+            raise exceptions.ValidationError("A network is required.")
+
+        # Note, we can't use the serializer's is_valid method here because that'll raise a 400 exception
+        # if the data is invalid (even if something exists). We need to catch that and raise a 409 instead.
+        name = request.data.get("name")
+        if not name:
+            raise exceptions.ValidationError("'name' is required.")
+
+        try:
+            network = Network.objects.get(network=network)
+        except Network.DoesNotExist:  # pragma: no cover
+            raise exceptions.NotFound("Network not found.")
+
+        # We do not have to worry about case sensitivity here, as the LowerCaseManager for the model will handle that.
+        try:
+            Community.objects.get(name=name, network=network)
+            raise ValidationError409(detail=f"Community with the name '{name}' already exists.")
```

Comment:
`raise` could be moved into `else` block to prevent subtle bugs stemming from raised error being caught by `except`. To be clear, it doesn't happen here, but should any of the exception definitions involved change, all bets are off.

Reply:
Thanks, refactored.